x-pack/legacy/plugins/index_management/public/components/mappings_editor/mappings_editor.tsx

```diff
@@ -20,17 +20,23 @@ interface Props {
   defaultValue?: { [key: string]: any };
 }
 
-export const MappingsEditor = React.memo(({ onUpdate, defaultValue = {} }: Props) => {
-  const configurationDefaultValue = Object.entries(defaultValue)
-    .filter(([key]) => CONFIGURATION_FIELDS.includes(key))
-    .reduce(
-      (acc, [key, value]) => ({
-        ...acc,
-        [key]: value,
-      }),
-      {} as Types['MappingsConfiguration']
-    );
-  const fieldsDefaultValue = defaultValue.properties || {};
+export const MappingsEditor = React.memo(({ onUpdate, defaultValue }: Props) => {
+  const configurationDefaultValue = useMemo(
+    () =>
+      defaultValue === undefined
+        ? ({} as Types['MappingsConfiguration'])
+        : Object.entries(defaultValue)
+            .filter(([key]) => CONFIGURATION_FIELDS.includes(key))
+            .reduce(
+              (acc, [key, value]) => ({
+                ...acc,
+                [key]: value,
+              }),
+              {} as Types['MappingsConfiguration']
```

Comment:
I'm also wondering if this namespacing of interfaces under `Types` helps. As a reader I find this quite a verbose way of specifying the interface we're targeting. If we want namespacing I'd prefer this pattern (using file as a namespace provider):

```ts
import * as types from './my_types'
...
const a: types.MappingsConfiguration = {};
```

Just my 2 cents :)

Reply:
I take note and will look how the exported types could be exported. It's mainly to avoid name conflict exporting and interface or a component from the same index.ts

But I am not sure that `types.MappingsConfiguration` is much less verbose than `Types['MappingsConfiguration']`. 😊